drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/control/DeleteColumnManagementAnchorWidget.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.guided.dtable.client.widget.table.columns.control;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import com.google.gwt.user.client.Command;
+import com.google.gwt.user.client.ui.Anchor;
+import org.drools.workbench.screens.guided.dtable.client.resources.i18n.GuidedDecisionTableConstants;
+import org.uberfire.ext.editor.commons.client.file.popups.DeletePopUpPresenter;
+
+@Dependent
+public class DeleteColumnManagementAnchorWidget extends Anchor {
+
+    private String columnHeader;
+    private Command afterDeleteConfirmedCommand;
+
+    @Inject
+    private DeletePopUpPresenter deletePopUpPresenter;
+
+    public DeleteColumnManagementAnchorWidget() {
+        super();
+        setText(GuidedDecisionTableConstants.INSTANCE.Delete());
+        setTitle(GuidedDecisionTableConstants.INSTANCE.DeleteThisColumn());
+        addClickHandler((clickEvent) -> {
+            deletePopUpPresenter.setPrompt(GuidedDecisionTableConstants.INSTANCE.DeleteColumnWarning(columnHeader));
```

Comment:
Wouldn't it be nice to have a test verifying you're hiding the comments box and changing the prompt?